sale_probability_amount/models/sale_order.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2023 Akretion (https://www.akretion.com).
+# @author Matthieu SAISON <matthieu.saison@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    probability = fields.Integer(
+        "Winning sale %", default=lambda self: self._get_default_value()
+    )
+
+    expected_amount_cur = fields.Monetary(
+        "Expected Amount in currency",
+        compute="_compute_expected_amount_cur",
+        currency_field="company_currency_id",
+        store=True,
+    )
+
+    def _get_default_value(self):
+        return (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("quotation_default_probability", 50)
+        )
+
+    def action_confirm(self):
+        res = super().action_confirm()
+        self.probability = 100
+        return res
```

Comment:
Just curious.  Was there a reason not to do:

```python
        self.probability = 100
        return super().action_confirm()
```

Reply:
sale_tier_validation don't pass the test if we set the probability before action_confirm. We can't change a sale under validation.